lib/rubocop/cop/lint/void.rb

```diff
@@ -106,13 +127,27 @@ def check_defined(node)
           add_offense(node, message: format(DEFINED_MSG, node.source))
         end
 
+        def check_nonmutating(node)
+          methods = [Array, String].flat_map { |c| nonmutating_methods(c) }
+          return unless node.send_type? && methods.include?(node.method_name)
+
+          add_offense(node, message: format(NONMUTATING_MSG, node.method_name,
+                                            node.method_name))
+        end
+
         def in_void_context?(node)
           parent = node.parent
 
           return false unless parent && parent.children.last == node
 
           VOID_CONTEXT_TYPES.include?(parent.type) && parent.void_context?
         end
+
+        def nonmutating_methods(klass)
```

Comment:
I think `[Array, String].flat_map{|c|nonmutating_methods(c)}` should be a constant for performance.